tools/cache_creator/units/cache_info_tests.cpp

```diff
@@ -0,0 +1,192 @@
+  CHECK(!consumeErrorToBool(privateHeaderInfoOrErr));
+  CHECK(privateHeaderInfoOrErr->privateHeader == privateHeader);
+  CHECK(privateHeaderInfoOrErr->contentBlobSize == 0);
+
+  auto contentOffsetOrErr = blobInfoOrErr->getCacheContentOffset();
+  CHECK(!consumeErrorToBool(contentOffsetOrErr));
+  CHECK(*contentOffsetOrErr == buffer.size());
+
+  // This should succeed as it is valid to have an empty cache content, i.e., zero entries.
+  llvm::SmallVector<cc::BinaryCacheEntryInfo, 0> entries;
+  auto err = blobInfoOrErr->readBinaryCacheEntriesInfo(entries);
+  CHECK(!consumeErrorToBool(std::move(err)));
+  CHECK(entries.empty());
+}
+
+TEST_CASE("Valid blob w/ one entry") {
+  const size_t trailingSpace = 16;
+  const size_t entrySize = sizeof(uint32_t);
+  const size_t bufferSize = vk::VkPipelineCacheHeaderDataSize + trailingSpace +
+                            sizeof(vk::PipelineBinaryCachePrivateHeader) + sizeof(vk::BinaryCacheEntry) + entrySize;
+  llvm::SmallVector<uint8_t> buffer(bufferSize);
+  uint8_t *currData = buffer.data();
+
+  auto *publicHeader = new (currData) vk::PipelineCacheHeaderData();
+  publicHeader->headerLength = vk::VkPipelineCacheHeaderDataSize + trailingSpace;
+  currData += publicHeader->headerLength;
+
+  auto *privateHeader = new (currData) vk::PipelineBinaryCachePrivateHeader();
+  currData += sizeof(vk::PipelineBinaryCachePrivateHeader);
+
+  auto *entryHeader = new (currData) vk::BinaryCacheEntry();
+  entryHeader->dataSize = entrySize;
+  currData += sizeof(vk::BinaryCacheEntry);
+
+  auto *entryContent = new (currData) uint32_t(42);
+
+  auto blobPtr = llvm::MemoryBuffer::getMemBuffer(llvm::toStringRef(buffer), "valid_one_entry", false);
+  assert(blobPtr);
+
+  auto blobInfoOrErr = cc::CacheBlobInfo::create(*blobPtr);
+  CHECK(!consumeErrorToBool(blobInfoOrErr));
+
+  // These should all succeed as both headers can be fully parsed and there's one valid entry.
+  auto publicHeaderInfoOrErr = blobInfoOrErr->readPublicVkHeaderInfo();
+  CHECK(!consumeErrorToBool(publicHeaderInfoOrErr));
+  CHECK(publicHeaderInfoOrErr->publicHeader == publicHeader);
+  CHECK(publicHeaderInfoOrErr->trailingSpaceBeforePrivateBlob == trailingSpace);
+
+  auto privateHeaderInfoOrErr = blobInfoOrErr->readBinaryCachePrivateHeaderInfo();
+  CHECK(!consumeErrorToBool(privateHeaderInfoOrErr));
+  CHECK(privateHeaderInfoOrErr->privateHeader == privateHeader);
+  CHECK(privateHeaderInfoOrErr->contentBlobSize == (sizeof(vk::BinaryCacheEntry) + entrySize));
+
+  llvm::SmallVector<cc::BinaryCacheEntryInfo, 1> entries;
+  auto err = blobInfoOrErr->readBinaryCacheEntriesInfo(entries);
+  CHECK(!consumeErrorToBool(std::move(err)));
+  CHECK(entries.size() == 1);
+
+  cc::BinaryCacheEntryInfo &entry = entries.front();
+  CHECK(entry.entryHeader == entryHeader);
+  CHECK(entry.idx == 0);
+  CHECK(entry.entryBlob.data() == reinterpret_cast<const uint8_t *>(entryContent));
+  CHECK(entry.entryBlob.size() == entrySize);
```

Comment:
Should you be checking `sourceElfBlobIdentifier` and the MD5 hash as well?  That could probably be checked if you have at least one test that had an `sourceElfMD5ToIdentifier`.

Reply:
These unit tests cover mainly the error handling and cache structure parsing part of the logic. More 'positive' tests will be added as LIT regression tests, where this will be exercised. But now I think that it won't take that much extra effort to have the MD5 logic validated in unit tests too.